schema/avro_builder.go

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hamba/avro/v2"
+)
+
+// Builder builds avro.RecordSchema instances and marshals them into JSON.
+// Builder accepts arguments for creating fields and creates them internally
+// (i.e. a user doesn't need to create the fields).
+// All errors will be returned as a joined error when marshaling the schema to JSON.
+type Builder struct {
```

Comment:
This is now called `schema.Builder`, I'd either move it under `avro` or name it `AvroBuilder`.